cmd/incusd/api_cluster_evacuation.go

```diff
@@ -161,9 +162,17 @@ func evacuateInstances(ctx context.Context, opts evacuateOpts) error {
 		return fmt.Errorf("Missing migration callback function")
 	}
 
-	metadata := make(map[string]any)
+	// Limit the number of concurrent evacuations to run at the same time
+	numParallelEvacs := max(runtime.NumCPU()/1, 1) // FIXME after testing, bump NumCPU division to 8 or 16
+
+	group, groupCtx := errgroup.WithContext(ctx)
+	group.SetLimit(numParallelEvacs)
 
 	for _, inst := range opts.instances {
+		// Accesses variables groupCtx, inst, and opts
+		group.Go(func() error {
+		metadata := make(map[string]any) // FIXME indent code block after review (makes diff noisy)
```

Comment:
Either indent or move to new function before merging (creates a lot of diff noise).